internal/venice-common/src/main/java/com/linkedin/venice/pushmonitor/ReplicaStatus.java

```diff
@@ -133,31 +133,55 @@ private void removeOldStatuses() {
 
     long completeCount = statusHistory.stream().filter(status -> status.getStatus() == COMPLETED).count();
     long incPushCount = statusHistory.stream().filter(status -> isIncrementalPushStatus(status.getStatus())).count();
-    Iterator<StatusSnapshot> itr = statusHistory.iterator();
+    long topicSwitchCount =
+        statusHistory.stream().filter(status -> status.getStatus() == TOPIC_SWITCH_RECEIVED).count();
+    long endOfPushCount = statusHistory.stream().filter(status -> status.getStatus() == END_OF_PUSH_RECEIVED).count();
```

Comment:
Can we only scan once to get the count so it improves the performance here?
For example
```
         List<Status> statuses = Arrays.asList(Status.OK, Status.NOT_OK, Status.UNKNOWN, Status.OK, Status.NOT_OK);

        Map<Status, Long> statusCounts = statuses.stream()
                .collect(Collectors.groupingBy(status -> status, Collectors.counting()));

        statusCounts.forEach((status, count) -> System.out.println(status + ": " + count));

```

Reply:
Can it work easily for `isIncrementalPushStatus`?

Reply:
Inc push count will then be checking SOIP count + EOIP count

Reply:
Sure, change to use a single loop to count the numbers.